### PR TITLE
crfs, stargz: support char, block, and fifo entries

### PR DESCRIPTION
**Pull request by bradfitz**

### CODE REVIEW

File: crfs.go [hunk 1]
```diff
@@ -147,6 +152,13 @@ var (
 	_ fspkg.HandleReader       = (*node)(nil)
 )
 
+// makedev packs a major and minor version how Linux expects.
```
Comment (jonjohnsonjr) [inline crfs.go]: At least to me, this is pretty arcane.

I went looking for an authoritative source to link out to and found [`golang/sys/unix.MkDev`](https://github.com/golang/sys/blob/master/unix/dev_linux.go) which [references glibc](https://sourceware.org/git/?p=glibc.git;a=blob;f=bits/sysmacros.h;h=226023de7e167f89962f8d6a446d9feabff35bda;hb=HEAD). I don't know if it's reasonable to use `unix.Mkdev` directly, but we could link out to it to as documentation.

File: crfs_test.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		{1, 256, 1048832},
+		{1, 257, 1048833},
+	}
+	for _, tt := range tests {
+		got := makedev(uint32(tt.major), uint32(tt.minor))
+		if got != tt.want {
+			t.Errorf("makedev(%v, %v) = %v; want %v", tt.major, tt.minor, got, tt.want)
+		}
+	}
+
```
Comment (jonjohnsonjr) [inline crfs_test.go]: nit: extra newline

Comment (bradfitz) [reply crfs_test.go]: Deleted whole file now.

**Comment by bradfitz** [comment]
PTAL.

**Pull request merged**